### PR TITLE
XW-85 fix: Delete option to appear when all subcategories are deleted

### PR DESCRIPTION
**Pull request by MHamzaBham**

### CODE REVIEW

File: public/assets/js/custom.js [hunk 1]
```diff
@@ -642,6 +642,15 @@ function deleteCategory(id) {
                             if (categoryElement.find('.subcategory-card').length === 0) { // if no subcategory
                                 categoryElement.parent().find('.category-title').find('i').remove();
                                 categoryElement.parent().find('.dropdown-toggler').css('cursor', '');
+                                if(parentId) {
+                                    categoryElement.parent().find('.dropdown-menu').append(`
+                                        <li onclick="deleteCategory(${parentId})">
+                                            <a class="dropdown-item py-1" href="#" style="font-size: 12px;">
```
Comment (mjawaids) [inline public/assets/js/custom.js]: No inline styles. Better to use classes and define them in `.css` files.

Comment (mjawaids) [reply public/assets/js/custom.js]: Also, this markup is duplicated in `subCategoryList.blade.php` file. That means if we ever have to change it we have to change it at two places and forgetting at one place could introduce issues. This has to be in one place and used wherever is needed. This is also a candidate for component if we were using React/Vue/Angular.

**Review by mjawaids — CHANGES_REQUESTED** [review]
What's the business logic here? Is parent category not allowed to be deleted if it has child categories? If so then this has to be changed so that if parent category is deleted, all its child categories should also be deleted, but warn user before deleting it.

File: public/assets/js/custom.js (hunk 1 above)
Comment (MHamzaBham) [reply public/assets/js/custom.js]: > No inline styles. Better to use classes and define them in `.css` files.

I have replaced the inline style with a bootstrap class

Comment (MHamzaBham) [reply public/assets/js/custom.js]: > Also, this markup is duplicated in `subCategoryList.blade.php` file. That means if we ever have to change it we have to change it at two places and forgetting at one place could introduce issues. This has to be in one place and used wherever is needed. This is also a candidate for component if we were using React/Vue/Angular.

For this, a new blade component for dropdown menu is made and called in both index.blade and subCategoryList.blade

**Review by mjawaids — APPROVED** [review]

**Pull request merged**